src/contexts/notification-context.tsx

```diff
@@ -0,0 +1,139 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import React, { createContext, useContext, useState, useEffect } from 'react';
+
+import {
+  NotificationSettings,
+  NotificationType,
+  NotificationPreference,
+} from '../types/notification.types';
+
+const DEFAULT_NOTIFICATION_PREFERENCE: NotificationPreference = {
+  enabled: true,
+  sound: true,
+  vibration: true,
+  importance: 'default',
+};
+
+const DEFAULT_NOTIFICATION_SETTINGS: NotificationSettings = {
+  masterToggle: true,
+  showInForeground: true,
+  preferences: {
+    message: { ...DEFAULT_NOTIFICATION_PREFERENCE, importance: 'high' },
+    task: { ...DEFAULT_NOTIFICATION_PREFERENCE, importance: 'high' },
+    alert: { ...DEFAULT_NOTIFICATION_PREFERENCE, importance: 'high' },
+    update: { ...DEFAULT_NOTIFICATION_PREFERENCE },
+    reminder: { ...DEFAULT_NOTIFICATION_PREFERENCE },
+    social: { ...DEFAULT_NOTIFICATION_PREFERENCE },
+  },
+};
+
+interface NotificationContextType {
+  settings: NotificationSettings;
+  isLoading: boolean;
+  updateMasterToggle: (enabled: boolean) => Promise<void>;
+  updateShowInForeground: (enabled: boolean) => Promise<void>;
+  updateTypePreference: (
+    type: NotificationType,
+    preference: Partial<NotificationPreference>,
+  ) => Promise<void>;
+  isNotificationTypeEnabled: (type: NotificationType) => boolean;
+}
+
+const NotificationContext = createContext<NotificationContextType | undefined>(undefined);
+
+const STORAGE_KEY = '@notification_settings';
+
+export const NotificationContextProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [settings, setSettings] = useState<NotificationSettings>(DEFAULT_NOTIFICATION_SETTINGS);
+  const [isLoading, setIsLoading] = useState(true);
+
+  // Load saved settings on mount
```

Comment:
Avoid comments like this, here and other places. See our guidance on comments:

[How to write better code comments](https://dev.to/adammc331/todo-write-a-better-comment-4c8c)